src/mlpack/methods/decision_tree/all_categorical_split_impl.hpp

```diff
@@ -30,6 +30,13 @@ double AllCategoricalSplit<FitnessFunction>::SplitIfBetter(
     arma::vec& splitInfo,
     AuxiliarySplitInfo& /* aux */)
 {
+  // Sanity check on data.
+  util::CheckSameElemSizes(data, labels, "AllCategoricalSplit()",
+  	"labels");
+  if(UseWeights)
+  util::CheckSameElemSizes(data, weights, "AllCategoricalSplit()",
+  	"weights");
```

Comment:
What's the problem with using `CheckSameSizes()`?  I'm hesitant to add new methods that do basically the same thing.